### DISCUSSION TITLE
Is adding array creation and manipulation routines a good idea?

### DISCUSSION THREAD

**Discussion by johnpalermo** [c128lib/.github#8] [Team Posts]
Should we add **array** routines to framework?
1) 1D arrays
2) 2D arrays
3 N dimensional arrays
4) Array creation
5) Array manipulation
etc.

**Comment by intoinside** [comment]
Yes, good, we need to discuss this argument because it may explode in complexity, it's a good idea!

**Comment by intoinside** [comment]
Just to mention something:
* array creation, initialization, merge...
* direct access (and set value) on specific entry
* operation on array like count entry, display on screen, sum, add/subtract/multiply/divide by constant value
* reverse element order
We should mention and then prioritize every operation

**Comment by johnpalermo** [comment]
Sounds good.  I could create the issues in framework as new enhancements.

**Comment by johnpalermo** [comment]
I'm starting to think about the array library again. I'm wondering if it's good to make a data structure to define a full array or leave some of the definition of the array up to the programmer.

A complete data structure would store in memory, along with the data, the size of the array for a 1D array and the number of rows and columns for a 2D array. This would be convenient for the programmer as he/she would not need to keep track of array sizes for each array separately. No need to pass the size of the array to the macro. This can go further and store the type of the array (1D/2D) in the data structure.

The alternative is to leave the definition and size of the array up to the programmer. Size would be passed to the macro. Programmer would need to keep track of the size of each array separately and call separate macros for 1D and 2D.

Welcoming your thoughts.

**Reply by intoinside** [reply]
I like your first approach, where programmer don't need to keep track of sizes. It gives a little abstraction and should simplify work
